thirdparty/CMakeLists.txt

```diff
@@ -329,7 +329,6 @@ ExternalProject_Add(curl
         --disable-smtp
         --disable-telnet
         --disable-tftp
-        --disable-shared
```

Comment:
That the libcurl is referenced by pegasus shell would lead to [hidden symbol XXX in YYY is referenced by DSO ...](https://stackoverflow.com/questions/23696585/what-does-exactly-the-warning-mean-about-hidden-symbol-being-referenced-by-dso):
![wecom-temp-75125-1ea1e87155e22755a232bec4998a343c](https://github.com/apache/incubator-pegasus/assets/743379/6f2aedf2-1d0f-4ffa-8bb6-55142986e6e4)